src/app/SIPUserAgents/SIPUserAgent.cs

```diff
@@ -1652,8 +1652,11 @@ private SIPRequest GetReferRequest(SIPDialogue target, string[] customHeaders)
         /// </summary>
         private void CallEnded()
         {
-            m_uac = null;
-            m_uas = null;
+            // SCB - Which call has ended?
+            // The UserAgent should be re-used for all calls. Only the sipdialogue associated with the call
+            // needs to be cleared up.
+            //m_uac = null;
+            //m_uas = null;
             m_callDescriptor = null;

```

Comment:
This change might be problematic. There's various logic that relies on checking m_uac or m_uas == null to determine whether there is an active all or not.

If it's a matter of macthing BYE requests to check if it's for the current call there's a better way. Logic could be added to DialogRequestReceivedAsync to check the Call-ID header.

Reply:
Now using callId to ensure that the relevant call is closed